lib/hermetic/slack.ex

```diff
@@ -23,10 +23,19 @@ defmodule Hermetic.Slack do
     ])
   end
 
+  @spec get_email(String.t()) :: String.t()
+  def get_email(userid) do
+    resp = HTTPoison.get!(@base_url <> "/users.profile.get?" <> URI.encode_query([
```

Comment:
I don't like adding payload decode in the generic request. In at least one instance I need to look at the headers rather than the body. Should I write a getJSON which wraps request?

Reply:
Use `HTTPoison.Base` instead: https://github.com/edgurgel/httpoison#wrapping-httpoisonbase

Reply:
Apparently, `HTTPoison.Base` is pretty limited in what it can do: It allows preprocessing the body and headers independently only, which means I can't have the body decoding  depend on the `Content-Type` in the headers.